ldap/servers/plugins/pwdstorage/gost_yescrypt.c

```diff
@@ -0,0 +1,66 @@
+/** BEGIN COPYRIGHT BLOCK
+ * License: GPL (version 3 or any later version).
+ * See LICENSE for details.
+ * END COPYRIGHT BLOCK **/
+
+#ifdef HAVE_CONFIG_H
+#include <config.h>
+#endif
+
+#include <crypt.h>
+#include <errno.h>
+
+#include "pwdstorage.h"
+
+int
+gost_yescrypt_pw_cmp(const char *userpwd, const char *dbpwd)
+{
+    /* return 0 If the passwords match, return 1 if passwords do not match. */
+    int rc = 1;
+    char *hash;
+    struct crypt_data output;
+
+    memset(&output, 0, sizeof(output));
+    hash = crypt_rn(userpwd, dbpwd, &output, (int) sizeof(output));
+    if (!hash) {
+        slapi_log_err(SLAPI_LOG_ERR, GOST_YESCRYPT_SCHEME_NAME,
+                      "Unable to hash userpwd value: %d\n", errno);
+        return rc;
+    }
+
+    if (slapi_ct_memcmp(hash, dbpwd, strlen(dbpwd)) == 0) {
+        rc = 0;
+    }
+
+    return rc;
+}
+
+char *
+gost_yescrypt_pw_enc(const char *pwd)
+{
+    const char *prefix = "$gy$";
+    char salt[CRYPT_GENSALT_OUTPUT_SIZE];
+    char *hash;
+    char *enc = NULL;
+    struct crypt_data output;
+
+    memset(&output, 0, sizeof(output));
```

Comment:
Minor comment...

Couldn't you do this instead:  struct crypt_data output = {0};

I don't think its any more efficient, but it's more consistent with other parts of our code.

Reply:
It is actually more efficient. I'm annoyed I missed this since I originally removed all uses of memset.